### ISSUE TITLE
interrupted download not resumed

### ISSUE THREAD

**Issue by pohly** [clearlinux/swupd-client#40]
swupd client 3.3.0
Python 2.7.9 SimpleHTTP as server

If I interrupt "swupd bundle-add full" ("full" being the bundle name) while it is downloading pack-full-from-0-to-41309480.tar, it leaves /var/lib/swupd/pack-full-from-0-to-41309480.tar behind.

Retrying the operation then fails:
```
# swupd bundle-add --url http://192.168.7.1:8000 full
swupd-client bundle adder 3.3.0
   Copyright (C) 2012-2016 Intel Corporation

Added bundle full for installation
Downloading required packs...
Attempting to download version string to memory
Downloading full pack for version 41309480
Curl error: 33 - see curl.h for details
pack downloads failed, cannot proceed with the installation, exiting.
Possible filedescriptor leak: fd_number="3",fd_details="/dev/urandom"
```

strace shows:
```
write(1, "Downloading full pack for versio"..., 43Downloading full pack for version 41309480
) = 43
lstat64("/var/lib/swupd/pack-full-from-0-to-41309480.tar", {st_mode=S_IFREG|0600, st_size=3718986, ...}) = 0
socket(PF_INET, SOCK_STREAM, IPPROTO_TCP) = 5
fcntl64(5, F_GETFL)                     = 0x2 (flags O_RDWR)
fcntl64(5, F_SETFL, O_RDWR|O_NONBLOCK)  = 0
connect(5, {sa_family=AF_INET, sin_port=htons(8000), sin_addr=inet_addr("192.168.7.1")}, 16) = -1 EINPROGRESS (Operation now in progress)
...
getpeername(5, {sa_family=AF_INET, sin_port=htons(8000), sin_addr=inet_addr("192.168.7.1")}, [16]) = 0
getsockname(5, {sa_family=AF_INET, sin_port=htons(36868), sin_addr=inet_addr("192.168.7.2")}, [16]) = 0
...
send(5, "GET /41309480/pack-full-from-0.t"..., 107, MSG_NOSIGNAL) = 107
...
recv(5, "HTTP/1.0 200 OK\r\nServer: SimpleH"..., 16384, 0) = 16384
...
close(5)                                = 0
write(1, "Curl error: 33 - see curl.h for "..., 40Curl error: 33 - see curl.h for details
) = 40
lstat64("/var/lib/swupd/pack-full-from-0-to-41309480.tar", {st_mode=S_IFREG|0600, st_size=3718986, ...}) = 0
write(1, "pack downloads failed, cannot pr"..., 70pack downloads failed, cannot proceed with the installation, exiting.
) = 70
close(4)                                = 0

```

I am using Python 2.7.9 SimpleHTTP as server (https://github.com/ostroproject/ostro-os/blob/master/scripts/swupd-http-server).

**Comment by phmccarty** [comment]
For pack downloads, swupd sets CURLOPT_RESUME_FROM_LARGE, so I expect the request failed because SimpleHTTP does not accept range requests. To resolve this issue, I think swupd should try downloading the /version/formatN/latest file with that curl option set, and if error 33 is returned, then do not set the option for any pack downloads.

Thanks for the report.

**Comment by fandeli** [comment]
Patch that fixes this issue has been sent for review.

**Comment by phmccarty** [comment]
This was fixed with #59
